java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -764,27 +764,23 @@ private void handleSubscribeChannels(ServerAction serverAction, JsonElement json
             serverAction.setResultMsg("Successfully applied state: " + ApplyStatesEventMessage.CHANNELS);
             SubscribeChannelsAction sca = (SubscribeChannelsAction)action;
 
-            // activate the new tokens
-            List<Long> newTokenIds = sca.getDetails().getAccessTokens()
-                    .stream()
-                    .filter(ac -> ac.getMinion() != null)
-                    .filter(ac -> ac.getMinion().getId().equals(serverAction.getServer().getId()))
-                    .map(AccessToken::getId)
-                    .collect(Collectors.toList());
-
             // if successful update channels in db and trigger pillar refresh
             SystemManager.updateServerChannels(
                     action.getSchedulerUser(),
                     serverAction.getServer(),
                     Optional.ofNullable(sca.getDetails().getBaseChannel()),
-                    sca.getDetails().getChannels(),
-                    newTokenIds);
+                    sca.getDetails().getChannels());
         }
         else {
-            Object returnObject = Json.GSON.fromJson(jsonResult, Object.class);
-            Gson gson = new GsonBuilder().setPrettyPrinting().create();
-            String json = gson.toJson(returnObject);
-            serverAction.setResultMsg("Failed to apply state: " + ApplyStatesEventMessage.CHANNELS);
+            //set the token as invalid
+            SubscribeChannelsAction sca = (SubscribeChannelsAction)action;
+            sca.getDetails().getAccessTokens().forEach(token -> {
+                token.setValid(false);
+                AccessTokenFactory.save(token);
+            });
+
+            serverAction.setResultMsg("Failed to apply state: " + ApplyStatesEventMessage.CHANNELS + ".\n" +
+                    jsonResult);
```

Comment:
isn't this jsonResult plain json and needs some formatting?

Reply:
I tried with `getJsonResultWithPrettyPrint` but the output result was exactly the same. You can see an example here:
![Screenshot from 2022-01-18 17-16-21](https://user-images.githubusercontent.com/4320859/149975851-8331d9f7-955b-4e3a-9df6-5b3f51e618d0.png)

Reply:
Uhhh, this is looking bad.

Can you try if this produce something more human readable?

```
statesResult.setOutput(YamlHelper.INSTANCE
                .dump(Json.GSON.fromJson(jsonResult, Object.class)).getBytes());
```

Reply:
we have a winner :+1:
![Screenshot from 2022-01-18 18-01-13](https://user-images.githubusercontent.com/4320859/149983700-ac3ac0a0-296b-4a86-ba2c-fc1fd16dee28.png). I'll substiture the content of `getJsonResultWithPrettyPrint` with this solution, and use it in the rest of the code

Reply:
Looks much better. I think we can improve more, but not in this PR.

Reply:
change `getJsonResultWithPrettyPrint` seems to be safe. Just to be sure I would not squash this PR (so we have 2 commit, one for the main changes and one for  `getJsonResultWithPrettyPrint` ). What do you think?

Reply:
Sounds good

Reply:
I try to subscribe channels when a minion is in blackout mode and yes, `getJsonResultWithPrettyPrint` slightly changed the String (the initial character is now `'` instead of `"` ) . I changed the `if` to match the new string